pkg/apis/networking/v1/defaults_test.go

```diff
@@ -234,6 +237,99 @@ func TestSetDefaultNetworkPolicy(t *testing.T) {
 	}
 }
 
+func TestSetDefaultsForIngressClassParametersReference(t *testing.T) {
+	tests := []struct {
+		original          *networkingv1.IngressClass
+		expected          *networkingv1.IngressClass
+		enableFeatureGate bool
+	}{
+		{ // Populated parameters should set the default Scope when feature is enabled
+			original: &networkingv1.IngressClass{
+				Spec: networkingv1.IngressClassSpec{
+					Controller: "controller",
+					Parameters: &networkingv1.IngressClassParametersReference{
+						Kind: "k",
+						Name: "n",
+					},
+				},
+			},
+			expected: &networkingv1.IngressClass{
+				Spec: networkingv1.IngressClassSpec{
+					Controller: "controller",
+					Parameters: &networkingv1.IngressClassParametersReference{
+						Kind:  "k",
+						Name:  "n",
+						Scope: networkingv1.IngressClassParametersReferenceScopeCluster,
+					},
+				},
+			},
+			enableFeatureGate: true,
+		},
+		{ // Empty Parameters should not set the default Scope when feature is enabled
```

Comment:
Need a test case that covers feature gate enabled + parameters specified with empty scope.

Reply:
Isn't that the first test case? 🤔

Reply:
Woops, you're completely right. I think the test case that's missing is one to ensure that the default does not override a specified scope. So if scope is set to namespace, the default should not override it.